post.js

```diff
@@ -1,10 +1,25 @@
 async function loadPost(url) {
   try {
-    const response = await fetch(url);
+    const headers = new Headers({
+      Accept: "application/activity+json, application/ld+json, text/html",
+    });
+
+    const response = await fetch(url, { headers });
     if (!response.ok) {
       throw new Error(`HTTP error! Status: ${response.status}`);
     }
-    return await response.text();
+
+    const contentType = response.headers.get("content-type");
+    if (
+      contentType.includes("application/ld+json") ||
+      contentType.includes("application/activity+json")
+    ) {
+      // Directly return JSON-LD if the response is JSON-LD or ActivityPub type
+      return await response.json();
+    } else {
+      // Return HTML content for further processing if the response is HTML
```

Comment:
Instead of returning the HTML, do the processing inside loadPost and look up the link from the header, then return loadpost on the linked URL

Reply:
Works with `ipns://` and `hyper://` but fails to load from `https://`

Error:
```
JSON-LD URL not found in the post
```
Commit:
https://github.com/hyphacoop/reader.distributed.press/pull/1/commits/7c8fd4b9a097be872d3e49d07ca33df062ec0ef2

Reply:
Nvm, added `https://` in `loadAndRenderPost`:
https://github.com/hyphacoop/reader.distributed.press/blob/3deb01090dabac4c7c0130b1bb4f3b17375f662b/post.js#L146